README.md

```diff
@@ -1,3 +1,171 @@
 # npecl design
 
-more info as it is written...
+A work in progress design documentation for the "new pecl" iteration.
+
+Basis:
+
+ * [Discussion doc](https://docs.google.com/document/d/1_N0E9xo3jn9aKrIZHIbTYaY5lXw71BpSO6-it4cRpDo/edit)
+ * Internal PHPF dev meet kick-off to review (minutes?)
+
+## Assumptions
+
+ * npecl is the assumed name, although it is not confirmed
+ * `npecl` is a CLI tool, probably bundled as a PHAR like Composer is
+   * therefore a usage such as `npecl install <thing>` assumes the PHAR exist in `$PATH`, for example in
+     `/usr/bin/npecl`. If not, you would do something like `php npecl.phar install <thing>`.
+
+## npecl itself
+
+ * use `symfony/console` to simplify writing the CLI tool itself
```

Comment:
Another potential option: https://github.com/minicli/minicli

Reply:
I'd say use symfony/console as Composer already depends on that..

Reply:
agree